apps/remote_control/lib/lexical/remote_control/search/store/mnesia.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Lexical.RemoteControl.Search.Store.Mnesia do
+  alias Lexical.Project
+  alias Lexical.RemoteControl.Search.Store.Backend
+  alias Lexical.RemoteControl.Search.Store.Mnesia.Query
+  alias Lexical.RemoteControl.Search.Store.Mnesia.Schema
+  alias Lexical.RemoteControl.Search.Store.Mnesia.State
+
+  use GenServer
+
+  @behaviour Backend
+
+  defdelegate delete_by_path(path), to: Query
+  defdelegate drop(), to: Query
+  defdelegate find_by_references(references, type, subtype), to: Query
+  defdelegate find_by_subject(subject, type, subtype), to: Query
+  defdelegate insert(entries), to: Query
+  defdelegate replace_all(entries), to: Query
+  defdelegate select_all(), to: Query
+
+  def new(%Project{} = project) do
+    with {:ok, _pid} <- start_link(project),
+         :ok <- wait_for_start() do
+      {:ok, Schema.load_state()}
+    end
+  end
+
+  def config do
+    Application.get_env(:remote_control, __MODULE__, [])
+  end
+
+  def set_persist_to_disc(persist?) do
+    config = Keyword.put(config(), :persist_to_disc, persist?)
+    Application.put_env(:remote_control, Mnesia, config)
+  end
+
+  def persist_to_disc? do
+    Keyword.get(config(), :persist_to_disc, true)
+  end
+
+  def sync(_) do
+    :ok
+  end
+
+  def destroy(_) do
+    Query.drop()
+    :ok
+  end
+
+  def wait_for_start do
+    GenServer.call(__MODULE__, :wait_for_start)
+  end
+
+  def initial_state do
+    GenServer.call(__MODULE__, :initial_state)
+  end
+
+  def start_link(%Project{} = project) do
+    GenServer.start_link(__MODULE__, [project], name: __MODULE__)
+  end
+
+  def init([%Project{} = project]) do
+    {:ok, State.new(project), {:continue, :connect}}
+  end
+
+  def handle_continue(:connect, %State{} = state) do
+    case State.connect(state) do
+      {:connect_to_node, new_state} ->
+        {:noreply, new_state, {:continue, :connect_to_node}}
+    end
+  end
```

Comment:
I think this should be `ensure_node` (mentioned elsewhere) and I also wonder if it should just return the new state instead of a tuple, since it only ever returns one thing.

```suggestion
  def handle_continue(:ensure_node, %State{} = state) do
    new_state = State.ensure_node(state)
    {:noreply, new_state, {:continue, :connect_to_node}}
  end
```